### DISCUSSION TITLE
Action Release trigger

### DISCUSSION THREAD

**Discussion by alygin** [zed-industries/zed#8310] [Show and tell]
Some handy code and workspace navigation features in other editors use shortcuts that are handled when keys are pressed and when they are released.

For instance, first `ctrl+tab` opens the tab switcher and automatically selects the previously opened tab, all the consecutive `ctrl+tab`s iterate selection through the list items, and when the `ctrl` key is released, we open the currently selected tab.

Recently, there were several disussions and requests for such behaviour:

- https://github.com/zed-industries/zed/issues/4450#issuecomment-1908621350
- #8258
- https://github.com/zed-industries/zed/pull/7987#issuecomment-1951717141

At the moment, GPUI doesn't provide API for implementing such navigation features.

My idea is to add `on_action_release()` / `capture_action_release()` to the GPUI elements in addition to `on_action()` / `capture_action()` that would trigger when all modifier keys are released right after an action has been performed.

Here's a PoC implementation: [Support for action release events](https://github.com/alygin/zed/pull/1/files) that allowed me to easily make [this Tab Switcher](https://github.com/zed-industries/zed/pull/7987) work the same way as in VSCode: it's opened with `ctrl+tab`, navigated with `ctrl` still pressed and `tab`s hitted, and finally switches to the selected tab when `ctrl` is released:

https://github.com/zed-industries/zed/assets/2101250/a3519fdb-8a2e-4aec-b19b-cc28d63bd26f

What do you think of such feature, of the proposed API and the way it's implemented?

**Comment by Moshyfawn** [comment]
I'm going to turn this into an issue, as there are several items being actively worked on.
